### PR TITLE
[TS] Add the adapter pattern

### PR DESCRIPTION
**Pull request by dvmhmdsd**
In this PR, I have created the `Adapter` pattern examples and added the usage examples and exercises.

### CODE REVIEW

File: typescript/adapter/temperature-date.ts
Comment (mahyoussef) [inline typescript/adapter/temperature-date.ts:1]: Thanks a lot! I have just a small edit here:
Can you rename this to TemperatureData, not Date? not to get people confused

File: typescript/adapter/exercises/JSON-report-adaptee.ts
Comment (mahyoussef) [inline typescript/adapter/exercises/JSON-report-adaptee.ts:1]: Can we rename this file as well to be json-report-adaptee? It's all small-case to adapt all naming convention 😍

File: typescript/adapter/exercises/JSON-report.ts
Comment (mahyoussef) [inline typescript/adapter/exercises/JSON-report.ts:1]: This file as well json small-case

**Comment by dvmhmdsd** [comment]
@mahyoussef Thanks for your comments. I've updated the PR so check please.

**Pull request merged**